### DISCUSSION TITLE
Page load event is not firing while testing, works fine when checking manually

### DISCUSSION THREAD

**Discussion by VitalyAdweek** [cypress-io/cypress#17812] [Questions and Help]
**Having the following error when visiting the page:**
<img width="441" alt="Screen Shot 2021-08-19 at 2 17 53 PM" src="https://user-images.githubusercontent.com/82416786/130122980-a6f23865-c509-421b-8c07-ce8c391f187a.png">

**But see window.onload event when pasting this code to a console of the browser ( I slow down the page load speed to have time to insert it before the page is loaded ), so I guess onLoad event is firing:**
```
window.onload = function(e)
{
    console.error('document.readyState is ready')
};
```
<img width="754" alt="Screen Shot 2021-08-19 at 2 27 43 PM" src="https://user-images.githubusercontent.com/82416786/130124266-820ecb30-857d-4730-9b92-b5209a589198.png">

**Here is the test code:**
```
Cypress.on('uncaught:exception', (err, runnable) => { return false })
describe('Verticals pages test', () => {
    it.only( '/', function() {
        cy.visit( 'https://www.adweek.com/');
    })
})
```

**Comment by portyu9** [comment]
Hi @VitalyAdweek 

`window.onload` itself is not the problem. `cy.visit()` already waits for the browser's `load` event and does not resolve until that event fires.

The important difference is that your manual test and the Cypress run are not necessarily completing the same resource-loading path. If a stylesheet, script, iframe, ad, or other resource remains pending/blocked during the Cypress run, the browser may never reach the `load` state that Cypress is waiting for.

I would compare the **Network** and **Console** output between the normal browser and Cypress run and look specifically for pending, failed, or missing resources.

Also, this does not help with the page-load timeout:

```js
Cypress.on('uncaught:exception', () => false)
```

That only suppresses uncaught application exceptions; it does not make `cy.visit()` ignore a missing `load` event. I would remove it while debugging so you can see the real application errors.

Increasing `pageLoadTimeout` only helps if the page eventually finishes loading. If it never does, identify the resource or application code preventing the load from completing and fix/stub that dependency instead.

You can also confirm Cypress's view of the lifecycle with:

```js
cy.on('window:before:load', () => {
  cy.log('before load')
})

cy.on('window:load', () => {
  cy.log('load fired')
})

cy.visit('https://www.adweek.com/')
```

If `window:before:load` fires but `window:load` does not, the next step is the Network/Console comparison—not adding an arbitrary wait.
